ui/redux/selectors/content.js

```diff
@@ -123,7 +123,21 @@ export const makeSelectNextUnplayedRecommended = (uri: string) =>
             continue;
           }
 
-          if (!history.some((item) => item.uri === recommendedForUri[i])) {
+          const recommendedUriInfo = parseURI(recommendedUri);
+
+          if (claimsByUri[uri] && claimsByUri[uri].claim_id === recommendedUriInfo.claimId) {
+            // Skip myself (same claim ID)
+            continue;
+          }
+
+          if (
+            !history.some((h) => {
+              const directMatch = h.uri === recommendedForUri[i];
+              const claimNameMatch = h.uri.includes(recommendedUriInfo.claimName);
+              const idMatch = claimsByUri[h.uri] && claimsByUri[h.uri].claim_id === recommendedUriInfo.claimId;
```

Comment:
Shouldn't this check work in all cases? If so do we need to check for a name match?

Reply:
`No change`

`claimsByUri` doesn't exist unless we resolve the entire history at startup (don't want to do that) -- it was added as a "try if possible" effort, primarily to handle odd URLs like

>_lbry://vacuum-tube-computer-p.09-–-building#5212bc8bc63c373e2bf1ebc5b765595ed7b6514d
lbry://vacuum-tube-computer-p-09-–-building#5212bc8bc63c373e2bf1ebc5b765595ed7b6514d_